src/Framework/MSBuildEventSource.cs

```diff
@@ -0,0 +1,465 @@
+        {
+            WriteEvent(3);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of the end of the specified event.
+        /// </summary>
+        [Event(4)]
+        public void BuildStop()
+        {
+            WriteEvent(4);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of information of how a project file built.
+        /// </summary>
+        [Event(5)]
+        public void BuildProjectStart()
+        {
+            WriteEvent(5);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of the end of the specified event.
+        /// </summary>
+        [Event(6)]
+        public void BuildProjectStop()
+        {
+            WriteEvent(6);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of information relevant to resolving references.
+        /// </summary>
+        [Event(7)]
+        public void ComputeClosureStart()
+        {
+            WriteEvent(7);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of the end of the specified event.
+        /// </summary>
+        [Event(8)]
+        public void ComputeClosureStop()
+        {
+            WriteEvent(8);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of information relevant to resolving conditionals.
+        /// </summary>
+        /// <param name="condition">The condition being evaluated.</param>
+        [Event(9)]
+        public void EvaluateConditionStart(string condition)
+        {
+            WriteEvent(9, condition);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of the end of the specified event.
+        /// </summary>
+        /// <param name="condition">The condition being evaluated.</param>
+        /// <param name="result">The result of evaluating the condition.</param>
+        [Event(10)]
+        public void EvaluateConditionStop(string condition, bool result)
+        {
+            WriteEvent(10, condition, result);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of how the project data was evaluated.
+        /// </summary>
+        /// <param name="projectFile">Relevant information about where in the run of the progam it is.</param>
+        [Event(11)]
+        public void EvaluateStart(string projectFile)
+        {
+            WriteEvent(11, projectFile);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of progress made on Evaluate.
+        /// </summary>
+        /// <param name="projectFile">Relevant information about where in the run of the progam it is.</param>
+        [Event(12)]
+        public void EvaluatePhase0Start(string projectFile)
+        {
+            WriteEvent(12, projectFile);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of progress made on Evaluate.
```

Comment:
I don't find many of the doc comments in this particularly enlightening. I'd consider just not including them; I'm not sure it's worth the effort to improve them.

Reply:
Although I'll admit I didn't spend too long trying to craft nice comments, and I agree it isn't worth the effort to do so, I do think they add something of value for a novice that might be hard to see for someone already familiar with how MSBuild works.

Reply:
I can remove the comments on stop events, though, since those are uniformly unhelpful.